app/services/merge_duplicate_patients.rb

```diff
@@ -0,0 +1,194 @@
+class MergeDuplicatePatients
+  def initialize(patients)
+    @patients = patients.sort_by(&:recorded_at)
+  end
+
+  def earliest_patient
+    @patients.first
+  end
+
+  def latest_patient
+    @patients.last
+  end
+
+  def merge
+    new_patient = ActiveRecord::Base.transaction do
+      new_patient = create_patient
+      create_prescription_drugs(new_patient)
+      create_medical_history(new_patient)
+      create_phone_numbers(new_patient)
+      create_patient_business_identifiers(new_patient)
+      create_encounters_and_observables(new_patient)
+      create_appointments(new_patient)
+      create_teleconsultations(new_patient)
+      mark_as_merged(new_patient)
+      new_patient.reload
+    end
+
+    Rails.logger.info "Merged patients #{@patients.pluck(:id)} into patient #{new_patient.id}"
```

Comment:
What happens when the transaction above is rolled back because of a problem? Does this line of code still execute, or does the process halt with an exception?

Reply:
Additionally, should we add a log _before_ we start the merge, so we can quickly identify which patients are involved if a merge _fails_?

Reply:
The process would halt with an exception. We are planning on capturing this like [here](https://github.com/simpledotorg/simple-server/pull/2104/files#diff-d0645edf5c6ace2983e2040dc09f061206d993d2aa4b60e13945dc13213f701aR15) and reporting it.